### PR TITLE
feat(datepicker): move attribute support to options

### PR DESCRIPTION
**Pull request by wesleycho**
- Move some attribute usage to datepickerOptions

This gets us most of the way with the second task in #5423

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
 
   if (optionsUsed) {
     [
+      'dateDisabled',
```
Comment (deeg) [inline src/datepicker/datepicker.js]: Does customClass not need to go here?

File: src/datepicker/docs/readme.md [hunk 2]
```diff
@@ -33,6 +33,8 @@ The datepicker has 3 modes:
 
   The supported options are:
 
+  - customClass
+  - dateDisabled
```
Comment (deeg) [inline src/datepicker/docs/readme.md]: Looks like missing datepickerMode from previous changes?

File: src/datepicker/datepicker.js (hunk 1 above)
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Good catch, need to add it

File: src/datepicker/docs/readme.md (hunk 2 above)
Comment (wesleycho) [inline src/datepicker/docs/readme.md]: Yep, need to add it.

**Comment by wesleycho** [comment]
Added updates, updated unit tests to use options.

**Comment by deeg** [comment]
This LGTM

**Pull request closed without merging**